### PR TITLE
use dex-contract utility functions

### LINKED ISSUE #184: Move getUnlimitedOrderAmounts into dex-contracts
This function is implemented in a few different places, but the most appropriate seems to the the contract source.

https://github.com/gnosis/dex-liquidity-provision/blob/52d5ba7cbf2e2b4b2cd5a31507e86825d081f979/scripts/utils/price_utils.js#L175-L191

### PR DESCRIPTION
**Pull request by bh2smith**
This PR uses the same functionality that has been deleted here and moved into dex-contracts (as exchange utils).

Note that this PR partially addresses #147 (in that it removes an import passing along web3, artifacts with it) and Closes #184

**Test Plan:** That all existing/remaining unit tests are passing.

### CODE REVIEW

File: scripts/synthetix/facilitate_trade.js [hunk 1]
```diff
@@ -1,9 +1,9 @@
 const { SynthetixJs } = require("synthetix-js")
 const ethers = require("ethers")
 const fetch = require("node-fetch")
+const exchangeUtils = require("@gnosis.pm/dex-contracts")
```
Comment (nlordell) [inline scripts/synthetix/facilitate_trade.js:4]: nit:
```suggestion
const { getUnlimitedOrderAmounts } = require("@gnosis.pm/dex-contracts")
```

File: scripts/utils/price_utils.js [hunk 2]
```diff
@@ -201,7 +155,7 @@ module.exports = function (web3 = web3, artifacts = artifacts) {
       return true
     }
 
-    const marketPrice = getUnitPrice(
+    const marketPrice = exchangeUtils.getUnitPrice(
```
Comment (nlordell) [inline scripts/utils/price_utils.js:158]: nit: The `getUnitPrice` should probably be imported directly from the contracts module:
```
const { getUnitPrice } = require("@gnosis.pm/dex-contracts")
```

And then:
```suggestion
    const marketPrice = getUnitPrice(
```

File: scripts/utils/trading_strategy_helpers.js [hunk 3]
```diff
@@ -3,9 +3,9 @@ module.exports = function (web3 = web3, artifacts = artifacts) {
   const BN = require("bn.js")
   const fs = require("fs")
   const Contract = require("@truffle/contract")
+  const exchangeUtils = require("@gnosis.pm/dex-contracts")
```
Comment (nlordell) [inline scripts/utils/trading_strategy_helpers.js:6]: Same comment as above.

File: test/verification_script.js [hunk 4]
```diff
@@ -1,13 +1,13 @@
 const BN = require("bn.js")
 const assert = require("assert")
 const Contract = require("@truffle/contract")
+const exchangeUtils = require("@gnosis.pm/dex-contracts")
```
Comment (nlordell) [inline test/verification_script.js:4]: Here too.

**Review by nlordell — COMMENTED** [review]
Is there any reason you prefer importing an object instead of using the `const { ... } = require("...")` import destructuring pattern? I feel like most places use the destructuring pattern and it would be nice to keep consistent.

File: package.json [hunk 5]
```diff
@@ -57,4 +57,4 @@
     "truffle": "^5.1.26",
     "truffle-hdwallet-provider": "^1.0.0-web3one.1"
   }
-}
+}
```
Comment (nlordell) [inline package.json:60]: nit: missing newline at the end of file.

**Review by nlordell — APPROVED** [review]
Looks good to me!

File: package.json (hunk 5 above)
Comment (bh2smith) [reply package.json:60]:
```suggestion
}

```

**Review by fedgiac — APPROVED** [review]
Nice, `dex-contracts` is definitely a better place for these functions.

File: scripts/synthetix/facilitate_trade.js [hunk 6]
```diff
@@ -80,12 +80,12 @@ module.exports = async (callback) => {
     const sUSDTosETHFee = parseFloat(snxjs.utils.formatEther(await snxjs.Exchanger.feeRateForExchange(sUSDKey, sETHKey)))
 
     // Compute buy-sell amounts based on unlimited orders with rates from above.
-    const [buyETHAmount, sellSUSDAmount] = getUnlimitedOrderAmounts(
-      formatedRate * (1 - sUSDTosETHFee),
+    const { base: sellSUSDAmount, quote: buyETHAmount } = getUnlimitedOrderAmounts(
+      1 / (formatedRate * (1 - sUSDTosETHFee)),
       sETH.decimals,
       sUSD.decimals
```
[the remark this answers is not in the document]
Comment (bh2smith) [reply scripts/synthetix/facilitate_trade.js:86]: Funny story, these were actually wrong before, but now correct.

**Pull request merged**